web/tests/proportion-field-service-spec.js

```diff
@@ -0,0 +1,123 @@
+describe('proportion-field service', function() {
+    beforeEach(function () {
+
+        angular.mock.module('incuna-surveys.proportion-field');
+
+        inject(function(proportionField) {
+            this.proportionField = proportionField;
+        });
+
+        this.fields = [
+            {label: 'One', required: false, id: '1-0'},
+            {label: 'Two', required: false, id: '1-1'},
+            {label: 'Three', required: false, id: '1-2'},
+            {label: 'Four', required: false, id: '1-3'}
+        ];
+
+        this.values = {
+            0: 10,
+            1: 20,
+            2: 40,
+            3: 30
```

Comment:
No but the system provides an object. Long term I think we should move away from array based referencing as it is not very scalable.